design-proposals/eim-modular-decomposition.md

```diff
@@ -0,0 +1,441 @@
+  secure boot, so that device deployments are protected from unauthorized access.
+- **Day-Two Device Lifecycle Management:** As an ISV or edge solution vendor, I want to manage day-two device
+  lifecycle operations, including immutable OS updates, firmware updates, and CVE remediation, so that edge devices
+  remain secure and up-to-date.
+- **Custom Hardware Resource Configuration:** As an ISV or edge solution vendor, I want to customize device
+  configuration for Intel CPU, GPU, and NPU resources during day-one lifecycle management, so that applications can
+  be allocated appropriate hardware resources.
+- **Reference API Integration:** As an ISV or edge solution vendor, I want to use reference APIs for higher-layer
+  services such as trusted compute and cluster orchestration, so that my solutions can integrate seamlessly with
+  existing platforms.
+- **Partner Vendor Orchestration Validation:** As an ISV or edge solution vendor, I want to validate partner vendor
+  orchestration layers against new Intel BIOS, firmware, CPU, and GPU platforms using EIM, so that device management
+  enablement can be shifted earlier in the development lifecycle.
+
+**Original Equipment manufacturer** can leverage as a reference EIM design and implementation of the following
+workflows
+
+- **Automated Edge Device Commissioning:** As an OEM, I want to orchestrate device onboarding, OS provisioning,
+  Kubernetes and application deployment across warehouse fleets with optional QA validation, so that production
+  readiness stays consistent at scale.
+- **Fleet-wide Firmware and OS Upgrades:** As an OEM, I want to run automated OS and firmware upgrades across field
+  fleets, so that devices remain current without manual intervention.
+- **Out-of-band Activation and Control:** As an OEM, I want to automate device activation and manage fleets using
+  Intel vPro AMT and ISM, so that field operations stay secure and responsive.
+
+**End customer or Systems integrator** can leverage the complete EIM stack as a reference achieve following
+workflows
+
+- **Multi-tenant Day-0 Onboarding and Provisioning:** As an end customer or systems integrator, I want to onboard
+  devices and provision any (not just Ubuntu or EMT) operating systems across tenants on-premises or in the cloud so
+  that deployments remain consistent from the start.
+- **Multi-tenant Day-1 Configuration and Operations:** As an end customer or systems integrator, I want to configure
+  and operate edge devices per tenant across on-prem and cloud environments so that ongoing management stays
+  streamlined and isolated.
+- **Multi-tenant Day-2 Lifecycle Management:** As an end customer or systems integrator, I want to manage device
+  updates and lifecycle tasks for each tenant wherever the solution is deployed so that fleets remain secure and
+  compliant.
+
+### Deliverables
+
+1. **Modular consumption** – Customers must be able to deploy the whole EIM stack or any subset of modules (for
```

Comment:
The modular decomposition poses the major challenge on EIM data modelling. Currently, the vertical features such as vPro or Day2 assume existence of Host/Instance database objects (and heavily rely on them). Also they require devices are provisioned via EIM before. IMO to achieve this you would need at least to:
- decouple database objects so that they're not strictly tightened
- support BYOD to register hosts for EIM management